service/eventer/github/client.go

```diff
@@ -130,7 +130,10 @@ func (e *Eventer) fetchNewDeploymentEvents(project, environment string, etagMap
 
 		deployments[index].Statuses = deploymentStatuses
 	}
-	deployments = e.filterDeploymentsWithoutStatuses(deployments)
+
+	if filterStatuses {
+		deployments = e.filterDeploymentsWithoutStatuses(deployments)
+	}
```

Comment:
When updating initially we do not want to filter the deployment events. In case the eventer starts or restarts it should ensure the latest state of the projects within the TPO. When we filter anything we risk to update the project to an undesired version.